ui/src/components/DenylistForm/index.jsx

```diff
@@ -0,0 +1,240 @@
+    /**
+     * A GraphQL address response. Not needed when adding
+     * a new address.
+     */
+    address,
+    /** Set to `true` when adding a new address. */
+    isNewaddress: bool,
+    /** Callback function fired when an address is created. */
+    onAddressAdd: func.isRequired,
+    /** Callback function fired when an address is deleted. */
+    onAddressDelete: func,
+    /** If true, form actions will be disabled. */
+    loading: bool,
+  };
+
+  static defaultProps = {
+    isNewAddress: false,
+    address: null,
+    loading: false,
+    onAddressDelete: null,
+  };
+
+  state = {
+    notificationType: '',
+    notificationAddress: '',
+    // eslint-disable-next-line react/no-unused-state
+    prevAddress: null,
+    validation: {
+      address: {
+        error: false,
+        message: '',
+      },
+    },
+  };
+
+  static getDerivedStateFromProps({ isNewAddress, address }, state) {
+    if (
+      isNewAddress ||
+      (state.notificationAddress && state.prevAddress === address)
+    ) {
+      return null;
+    }
+
+    return {
+      notificationType: address.notificationType,
+      notificationAddress: address.notificationAddress,
+      prevAddress: address,
+      validation: {
+        address: {
+          error: false,
+          message: '',
+        },
+      },
+    };
+  }
+
+  handleDeleteAddress = () => {
+    const { notificationType, notificationAddress } = this.state;
+
+    this.props.onAddressDelete(notificationType, notificationAddress);
+  };
+
+  handleInputChange = ({ target: { name, value } }) => {
+    this.setState({ [name]: value });
+
+    if (
+      name === 'notificationType' &&
+      value !== DENYLIST_NOTIFICATION_TYPES.EMAIL
+    ) {
+      this.setState({
+        validation: {
+          address: {
+            error: false,
+            message: ' ',
+          },
+        },
+      });
+    }
+  };
+
+  handleAddressChange = e => {
+    this.setState({
+      notificationAddress: e.currentTarget.value,
+      validation: {
+        address: {
+          error: !e.currentTarget.validity.valid,
+          message: e.currentTarget.validationMessage,
+        },
+      },
+    });
+  };
+
+  handleAddAddress = () => {
+    const { notificationType, notificationAddress } = this.state;
+
+    this.props.onAddressAdd(notificationType, notificationAddress);
+  };
+
+  isFormValid = () => {
+    const { notificationType, notificationAddress, validation } = this.state;
+    const hasEmptyFields = !notificationAddress || !notificationType;
+    const isAddressValid = !validation.address.error;
+
+    return !hasEmptyFields && isAddressValid;
+  };
+
+  prettify = str =>
+    // remove underscores and capitalize first alphabet
+    str
+      .split('_')
+      .map(word => {
+        const pretty =
+          word !== 'IRC'
+            ? word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
+            : word;
+
+        return pretty;
+      })
+      .join(' ');
```

Comment:
You can make use of https://github.com/blakeembrey/change-case#readme to do things like remove underscores.